agent/external.go

```diff
@@ -0,0 +1,74 @@
+package agent
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"os/user"
+
+	log "github.com/sirupsen/logrus"
+
+	"golang.org/x/crypto/ssh/agent"
+	"golang.org/x/exp/slices"
+)
+
+var sshKeyTypeMap = map[string]string{
+	"rsa":     "ssh-rsa",
+	"ec":      "ssh-ecdsa",
+	"ed25519": "ssh-ed25519",
+}
+
+var supportedKeyTypes = []string{
```

Comment:
Minor suggestion:
```go
type keyType string
const (
    rsa keyType = "ssh-rsa"
    ecdsa           = "ssh-ecdsa"
    ed25519      = "ssh-ed25519"
)
```

Which would then allow a DRYer approach since you could re-use these values in the above map:
```go
var sshKeyTypeMap = map[string]keyType{
	"rsa":    rsa,
	"ec":      ecdsa,
	"ed25519": ed25519,
}
```
If you wish to conserve the `string` type properties - the `keyType` can become an alias instead of a sub-type. For generics, `~string`, will still work without aliasing.

Reply:
I was also pondering whether to make this a `map[string][]string` to support the various `-sk` sub-types that would also be acceptable matches in "best" selection. Hmm.